src/ui/CSimpleEditBox.cpp

```diff
@@ -768,6 +768,13 @@ void CSimpleEditBox::OnEscapePressed() {
     // TODO
 }
 
+void CSimpleEditBox::OnTabPressed() {
+    if (this->m_onTabPressed.luaRef) {
+        this->RunScript(this->m_onTabPressed, 0, nullptr);
+    }
```

Comment:
Let's put this part specifically in yet another function: `CSimpleEditBox::RunOnTabPressedScript`. I know it feels speculative for the 3.3.5a client, but 4.1.0 and 6.0.1 both have some surviving examples of `Run<script-name-here>Script` member functions.